### ISSUE TITLE
Reflection warnings

### ISSUE THREAD

**Issue by ikitommi** [lambdaisland/deep-diff#5]
on:

```clj
[lambdaisland/deep-diff "0.0-25"]
```

some:

```
Reflection warning, lambdaisland/deep_diff/diff.clj:103:15 - reference to field getName can't be resolved.
Reflection warning, lambdaisland/deep_diff/diff.clj:146:34 - reference to field getClass can't be resolved.
Reflection warning, lambdaisland/deep_diff/diff.clj:146:24 - reference to field isArray can't be resolved.
Reflection warning, lambdaisland/deep_diff/printer.clj:80:27 - reference to field getName can't be resolved.
```

**Comment by plexus** [comment]
Thanks for the report @ikitommi, seems most (or all?) of these could be fixed by typing hinting to `java.lang.Class`. Any chance you'd have a moment to make a PR?

**Comment by sogaiu** [comment]
How does this look?

```
diff --git a/src/lambdaisland/deep_diff/diff.clj b/src/lambdaisland/deep_diff/diff.clj
index 6ecda91..cf70cf2 100644
--- a/src/lambdaisland/deep_diff/diff.clj
+++ b/src/lambdaisland/deep_diff/diff.clj
@@ -100,7 +100,7 @@
 (defn- val-type [val]
   (let [t (type val)]
     (if (class? t)
-      (symbol (.getName t))
+      (symbol (.getName ^Class t))
       t)))
 
 (defn- diff-map [exp act]
@@ -143,7 +143,7 @@
 (extend Object
   Diff
   {:diff-similar (fn [exp act]
-                   (if (.isArray (.getClass exp))
+                   (if (.isArray (.getClass ^Object exp))
                      (diff-seq exp act)
                      (diff-atom exp act)))})
 
diff --git a/src/lambdaisland/deep_diff/printer.clj b/src/lambdaisland/deep_diff/printer.clj
index 0e722d6..95d303f 100644
--- a/src/lambdaisland/deep_diff/printer.clj
+++ b/src/lambdaisland/deep_diff/printer.clj
@@ -75,7 +75,7 @@
             (puget/format-doc printer v)]))))})
 
 (defn- print-handler-resolver [extra-handlers]
-  (fn [klz]
+  (fn [^Class klz]
     (and klz (get (merge @#'print-handlers extra-handlers)
                   (symbol (.getName klz))))))
```

**Comment by plexus** [comment]
Thanks @sogaiu, that looks good, would you mind submitting this as a PR?

**Comment by sogaiu** [comment]
It has been attempted :)

**Comment by plexus** [comment]
Released as `0.0-35`. Ping @ikitommi .

**Comment by ikitommi** [comment]
Thanks!!
